### ISSUE TITLE
Don't built all images every time

### ISSUE THREAD

**Issue by rabernat** [pangeo-data/pangeo-stacks#16]
I noticed that our travis config just builds all the images all the time.

We could replace this with the logic that is in hubploy to check based on commit whether a build is needed.

**Comment by jhamman** [comment]
Yes, we can do this. We basically just need to port the logic from hubploy that looks at the commit range for each image.
